usaspending_api/etl/management/commands/load_usaspending_assistance.py

```diff
@@ -168,16 +168,17 @@ def get_or_create_recipient(self, row):
                 row,
                 mapper=location_mapper_fin_assistance_recipient).location_id,
             "recipient_name": row['recipient_name'],
-            "recipient_unique_id": row['duns_no'],
         }
 
         recipient_type = row.get("recipient_type", ":").split(":")[1].strip()
         recipient_dict.update(self.recipient_flags_by_type(recipient_type))
 
-        le = LegalEntity.objects.filter(
-            recipient_unique_id=row['duns_no']).first()
-        if not le:
-            le = LegalEntity.objects.create(**recipient_dict)
+        le, created = LegalEntity.get_or_create_by_duns(duns=row['duns_no'])
+        if created:
+            # Update from our recipient dictionary
+            for attr, value in recipient_dict.items():
```

Comment:
I'm not familiar with the different types of recipient info we get from a historic data submission vs a broker submission. Am I interpreting correctly that here we're adding additional attributes to a new LegalEntity (e.g., name, recipient flags by type) that we don't add when creating a new LegalEntity via broke data?